branchLayoutImpl/src/main/java/com/virtuslab/branchlayout/impl/manager/BranchLayoutManager.java

```diff
@@ -0,0 +1,105 @@
+package com.virtuslab.branchlayout.impl.manager;
+
+import java.nio.file.Path;
+
+import io.vavr.Tuple;
+import io.vavr.Tuple2;
+import io.vavr.collection.List;
+import io.vavr.control.Try;
+import org.checkerframework.checker.index.qual.Positive;
+import org.checkerframework.checker.initialization.qual.UnderInitialization;
+
+import com.virtuslab.branchlayout.api.manager.IBranchLayoutManager;
+import com.virtuslab.branchlayout.api.manager.IBranchLayoutReader;
+import com.virtuslab.branchlayout.api.manager.IBranchLayoutWriter;
+import com.virtuslab.branchlayout.api.manager.IIndentDefining;
+import com.virtuslab.logger.IPrefixedLambdaLogger;
+import com.virtuslab.logger.PrefixedLambdaLoggerFactory;
+
+public class BranchLayoutManager implements IBranchLayoutManager {
+  private static final IPrefixedLambdaLogger LOG = PrefixedLambdaLoggerFactory.getLogger("branchLayout");
+
+  private final IBranchLayoutReader reader;
+  private final IBranchLayoutWriter writer;
+
+  public BranchLayoutManager(Path path, boolean isBranchLayoutPresent) {
+    char indentCharacter = BranchLayoutFileUtils.DEFAULT_INDENT_CHARACTER;
+    int indentWidth = BranchLayoutFileUtils.DEFAULT_INDENT_WIDTH;
+    if (isBranchLayoutPresent) {
+      IndentDerivator indentDerivator = new IndentDerivator(path);
+      indentCharacter = indentDerivator.getIndentCharacter();
+      indentWidth = indentDerivator.getIndentWidth();
+    }
+    this.reader = new BranchLayoutFileReader(path, indentCharacter, indentWidth);
+    this.writer = new BranchLayoutFileWriter(path, indentCharacter, indentWidth);
+  }
+
+  @Override
+  public IBranchLayoutReader getReader() {
+    return reader;
+  }
+
+  @Override
+  public IBranchLayoutWriter getWriter() {
+    return writer;
+  }
+
+  private static class IndentDerivator implements IIndentDefining {
+
+    private final char indentCharacter;
+    @Positive
+    private final int indentWidth;
+
+    IndentDerivator(Path path) {
```

Comment:
I think it's gonna be more natural if all that logic is placed directly in `BranchLayoutManager`

Reply:
And `indentCharacter`/`indentWidth` are stored in a simple `@Data` class with no logic